src/main/java/seedu/address/logic/commands/DeleteWeddingCommand.java

```diff
@@ -0,0 +1,78 @@
+package seedu.address.logic.commands;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_WEDDING;
+
+import java.util.List;
+
+import seedu.address.commons.util.ToStringBuilder;
+import seedu.address.logic.Messages;
+import seedu.address.logic.commands.exceptions.CommandException;
+import seedu.address.model.Model;
+import seedu.address.model.wedding.Wedding;
+
+/**
+ * Deletes a wedding identified using its wedding name in the Wedlinker.
+ */
+public class DeleteWeddingCommand extends Command {
+    public static final String COMMAND_WORD = "delete-wedding";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD
+            + ": Deletes the wedding identified by the wedding name.\n"
+            + "Parameters: WEDDING_NAME (must exist in the Wedlinker)\n"
+            + "Example: " + COMMAND_WORD + " "
+            + PREFIX_WEDDING + " Timothy's Wedding";
+
+    public static final String MESSAGE_DELETE_WEDDING_SUCCESS = "Deleted Wedding: %1$s";
+    public static final String MESSAGE_DELETE_WEDDING_FAILURE_STILL_USED = "The Wedding: %1$s is still used";
+    public static final String MESSAGE_DELETE_WEDDING_FAILURE_NOT_FOUND = "The Wedding: %1$s does not exist";
+
+    private final Wedding targetWedding;
+
+    public DeleteWeddingCommand(Wedding wedding) {
+        targetWedding = wedding;
+    }
+
+    @Override
+    public CommandResult execute(Model model) throws CommandException {
+        requireNonNull(model);
+        List<Wedding> allWeddings = model.getFilteredWeddingList();
+
+        for (Wedding wedding : allWeddings) {
+            if (wedding.getWeddingName().equals(targetWedding.getWeddingName())) {
+                if (wedding.canBeDeleted()) {
+                    model.deleteWedding(wedding);
+                    return new CommandResult(String.format(MESSAGE_DELETE_WEDDING_SUCCESS,
+                            Messages.format(targetWedding)));
+                } else {
+                    throw new CommandException(
+                            String.format(MESSAGE_DELETE_WEDDING_FAILURE_STILL_USED, Messages.format(targetWedding)));
```

Comment:
CodeCov flagged this as not covered -- also couldn't find any tests that had `MESSAGE_DELETE_WEDDING_FAILURE_STILL_USED` as output, but maybe I overlooked!